lib/dal.js

```diff
@@ -0,0 +1,86 @@
+var fs = require('fs');
+var Q = require('q');
+
+/** Reads from the configuration file and returns data
+ * @param {string} confFile - the path and filename to the configuration file
+ * @return {Q} - Q promise object
+ */
+function getConfig(confFile) {
+  var readJSON = Q.denodeify(fs.readFile.bind(fs.readFile));
+  return readJSON(confFile, 'utf8')
+    .then(function (data) {
+      var JSONdata = JSON.parse(data);
+      return Q.resolve(JSONdata);
+    });
+}
+
+/** Attempts to access configuration file data for the shared extension
+ * @param {string} confFile - the path and filename to the configuration file
+ * @param {string} name - the name of the sharedExtension to access
+ * @return {Q} - Q promise object
+ */
+
+var getSharedExtension = function(confFile, name) {
+  var defer = Q.defer();
+  var sharedExtension = {};
+  sharedExtension.pos = -1;
+  return getConfig(confFile).then(function(data) {
```

Comment:
You are putting an implementation detail directly in the API you expose. If you need to add database support later, you'll have a lot more work to separate this out from the functionality of returning a shared extension object. Also, what if you wanted to support both config files and a database?

You should have an internal only API for getting the data that can be modified to support various back-ends instead of having this inside the API you expose.